teuthology/provision/openstack.py

```diff
@@ -66,7 +66,10 @@ def attach_volumes(self, name, volumes):
             except subprocess.CalledProcessError as e:
                 if 'No volume with a name or ID' not in e.output:
                     raise e
-                self.run("volume create -f json " +
+                # do not use OpenStack().run because its
+                # bugous for volume create as of openstackclient 3.2.0
+                # https://bugs.launchpad.net/python-openstackclient/+bug/1619726
+                misc.sh("volume create -f json " +
```

Comment:
yes, good catch ! I repushed after testing and also added a try-catch to cope with the (rare) case when a volume does not even exist at all after volume create returns.